hal_st/stm32fxxx/DmaStm.cpp

```diff
@@ -57,151 +58,219 @@ namespace hal
 
 #if defined(DMA_CHANNEL_BASED)
         // clang-format off
-        const std::array dmaISR
+        const std::array dmaISRDma1
         {
-            std::array
-            {
-                &DMA1->ISR,
-                &DMA1->ISR,
-                &DMA1->ISR,
-                &DMA1->ISR,
-                &DMA1->ISR,
-                &DMA1->ISR,
+#if defined(STM32WBA)
+            &GPDMA1_Channel0_NS->CSR,
+            &GPDMA1_Channel0_NS->CSR,
+            &GPDMA1_Channel0_NS->CSR,
+            &GPDMA1_Channel0_NS->CSR,
+            &GPDMA1_Channel0_NS->CSR,
+            &GPDMA1_Channel0_NS->CSR,
+            &GPDMA1_Channel0_NS->CSR,
+            &GPDMA1_Channel0_NS->CSR,
```

Comment:
is the CSR register the same for all channels? Or is there a typo here that it should have said Channel0, Channel1 etc.?

Reply:
Good catch! It's a typo, fixed!